packages/manager/modules/overthebox/src/overthebox/migration/offers/overTheBox-migration-offers.controller.js

```diff
@@ -0,0 +1,103 @@
+import set from 'lodash/set';
+
+export default class OverTheBoxMigrationOffersCtrl {
+  /* @ngInject */
+  constructor(
+    $q,
+    $scope,
+    $translate,
+    TucToast,
+    OverTheBoxMigrationService,
+    OvhApiOverTheBox,
+  ) {
+    this.$q = $q;
+    this.$scope = $scope;
+    this.$translate = $translate;
+    this.TucToast = TucToast;
+    this.OverTheBoxMigrationService = OverTheBoxMigrationService;
+    this.OvhApiOverTheBox = OvhApiOverTheBox;
+  }
+
+  $onInit() {
+    this.offers = [];
+    this.loading = true;
+
+    this.$q.all([this.getCurrentOffer(), this.getOffers()]).finally(() => {
+      this.loading = false;
+    });
+  }
+
+  getCurrentOffer() {
+    return this.OvhApiOverTheBox.v6()
+      .get({
+        serviceName: this.serviceName,
+      })
+      .$promise.then((otb) => {
+        this.currentOffer = {
+          prettyOfferName: otb.prettyOfferName,
+          offer: otb.offer,
+        };
+        return otb;
+      });
+  }
+
+  getOffers() {
+    return this.OverTheBoxMigrationService.getOffers(this.serviceName)
+      .then(({ data }) => {
+        this.offers = data;
+        if (this.offers.length > 0) {
+          // Retrieve current offer price
+          this.currentOffer.currentOfferPrice = this.offers[0].currentOfferPrice;
+        }
+
+        // Build hardware options list
+        this.offers = this.offers.map((offer) => {
+          const hardwareOptions = [];
+
+          // Set hardware option return by API
+          let optionLabel = this.$translate.instant(
+            'overthebox_migration_hardware_yes',
+            { price: offer.hardwarePrice ? offer.hardwarePrice.text : '' },
```

Comment:
```suggestion
            { price: offer.hardwarePrice?.text ?? '' },
```

Reply:
I would like to do what you write but it's generated errors 
"Uncaught (in promise) Error: Module build failed (from ../../tools/webpack-config/src/loaders/translation-ui-router.js):
SyntaxError: Optional chaining cannot appear in left-hand side (49:21)"
So I can't do what you ask to do.